bootstrap_cfn/config.py

```diff
@@ -525,6 +526,8 @@ def rds(self, template):
             ],
             VpcId=Ref("VPC"),
             GroupDescription="SG for EC2 Access to RDS",
+            # translates to DependsOn="AttachGateway",
+            DependsOn=[r.title for r in self._find_resources(template, "AWS::EC2::VPCGatewayAttachment")],
```

Comment:
Both of these calls use the `template` object as an argument, which is passed when each template gets populated with resources above eg:

````
def elb(self, template):
```
and
```
def rds(self, template):
```
and during construction we create and add these resources, but it is not mandatory that the resources would always exist. During self.process() they exist because of this: 
```
    def process(self):
        template = self.base_template()

        vpc = self.vpc()
        map(template.add_resource, vpc)

        iam = self.iam()
        map(template.add_resource, iam)

        ec2 = self.ec2()
        map(template.add_resource, ec2)
```
...but for example during 
https://github.com/ministryofjustice/bootstrap-cfn/blob/fils-dependency-implementation/tests/tests.py#L292
..this no longer applies because no vpc has been populated (as it has been populated here: https://github.com/ministryofjustice/bootstrap-cfn/blob/fils-dependency-implementation/tests/tests.py#L376-L389 ).

We could make this a more generic function (and I was looking for one, which is how I found `_find_resources()` ) if you think it's clearer but it felt redundant to me. Or I can just split this to multiple lines as before.